File: but it keeps getting faster/faster.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(){

    string W;
    int N;

    cin >> W >> N;

    int time = 0;
    int speed = 1;
    int tmp = 0;

    string w;

    for (int i = 0; i < N;){
        for (; i < N && tmp > 0; i++, tmp--){
            cin >> w;
            if (w == W){
                speed++;
            }
        }
        time++;
        tmp = speed;
    }
    cout << time - 1 << "\n";

    return 0;
}

File: but it keeps getting faster/faster.py
#Scrivi qui il tuo codice

import math

W = str(input().strip())
N = int(input().strip())

C = [""] * N

for i in range(N):
    C[i] = str(input().strip())

time = 0
speed = 1

count = 0

for i in range(N):
    time += 1 / speed
    
    #print(time, speed, count)
    time = round(time, 3)
    if time == float(round(time)):
        time = round(time, 3)
        speed += count
        count = 0
        #print("pizza")
    
    #print(C[i], W, C[i] == W)
    if C[i] == W:
        count += 1
    

print(math.floor(time))
